pkgs/applications/misc/waybar/default.nix

```diff
@@ -65,6 +66,12 @@ stdenv.mkDerivation rec {
     )
   '';
 
+  # Patch for workspaces support in wlr/workspaces
+  # See https://wiki.hyprland.org/Useful-Utilities/Status-Bars/#waybar
+  postPatch = lib.optionalString hyprlandSupport ''
+    sed -i -e 's/zext_workspace_handle_v1_activate(workspace_handle_);/const std::string command = "hyprctl dispatch workspace " + name_;\n\tsystem(command.c_str());/g' src/modules/wlr/workspace_manager.cpp
```

Comment:
WTH?

Write a patch file instead.

Reply:
patch files work with changing locations of lines of code

Reply:
To an extent... a patch file is just a git diff and anything that can cause conflicts in a git diff will cause a patch to fail. I see 4997 usages of postPatch with the same intent of trying to regex catch and replace things. I have an upcoming PR that requires some of this postPatch regex logic utilizing substituteInPlace so I'm just trying to figure out why it's 'wrong'.

Reply:
Added a patch

Reply:
> anything that can cause conflicts in a git diff will cause a patch to fail.

That is precisely the idea.
You, or for that matter @jtbx, is manipulating a piece of C++ code. 
~~If~~ _When_ the upstream changes the code, this sed one-liner will not fail and silently continue, basically acting as a NOOP. Certainly this is not intended, and will be a source of heisenbugs.

On the other hand the patch file **will** fail, forcing him (or anyone that updates the package) to review the code.

Also, this is not really a sed one liner - it changes one line by other two. It looks like a job for a true patch file.

Reply:
Alright, that makes more sense. Thanks for the clarity. I was thinking about it a bit more since my comment and that was something that popped into my head.